include/boost/beast/http/impl/read.hpp

```diff
@@ -250,6 +144,225 @@ struct run_read_msg_op
+{
+    AsyncReadStream& s_;
+    DynamicBuffer& b_;
+    basic_parser<isRequest>& p_;
+    std::size_t bytes_transferred_;
+    bool cont_;
+
+public:
+    read_some_op(
+        AsyncReadStream& s,
+        DynamicBuffer& b,
+        basic_parser<isRequest>& p)
+        : s_(s)
+        , b_(b)
+        , p_(p)
+        , bytes_transferred_(0)
+        , cont_(false)
+    {
+    }
+
+    template<class Self>
+    void operator()(
+        Self& self,
+        error_code ec = {},
+        std::size_t bytes_transferred = 0)
+    {
+        BOOST_ASIO_CORO_REENTER(*this)
+        {
+            if(b_.size() == 0)
+                goto do_read;
+            for(;;)
+            {
+                // parse
+                {
+                    auto const used = p_.put(b_.data(), ec);
+                    bytes_transferred_ += used;
+                    b_.consume(used);
+                }
+                if(ec != http::error::need_more)
+                    break;
+
+            do_read:
+                BOOST_ASIO_CORO_YIELD
+                {
+                    cont_ = true;
+                    // VFALCO This was read_size_or_throw
+                    auto const size = read_size(b_, 65536);
+                    if(size == 0)
+                    {
+                        ec = error::buffer_overflow;
+                        goto upcall;
+                    }
+                    auto const mb =
+                        beast::detail::dynamic_buffer_prepare(
+                            b_, size, ec, error::buffer_overflow);
+                    if(ec)
+                        goto upcall;
+                    s_.async_read_some(*mb, std::move(self));
+                }
+                b_.commit(bytes_transferred);
+                if(ec == net::error::eof)
+                {
+                    BOOST_ASSERT(bytes_transferred == 0);
+                    if(p_.got_some())
+                    {
+                        // caller sees EOF on next read
+                        ec.assign(0, ec.category());
+                        p_.put_eof(ec);
+                        if(ec)
+                            goto upcall;
+                        BOOST_ASSERT(p_.is_done());
+                        goto upcall;
+                    }
+                    ec = error::end_of_stream;
+                    break;
+                }
+                if(ec)
+                    break;
+            }
+
+        upcall:
+            if(! cont_)
+            {
+                BOOST_ASIO_CORO_YIELD
+                    net::post(
+                        beast::bind_front_handler(std::move(self), ec));
+            }
+            self.complete(ec, bytes_transferred_);
+        }
+    }
+};
+
+template<class Stream, class DynamicBuffer, bool isRequest, class Condition>
+class read_op
+    : asio::coroutine
+{
+    Stream& s_;
+    DynamicBuffer& b_;
+    basic_parser<isRequest>& p_;
+    std::size_t bytes_transferred_;
```

Comment:
Why didn't you just initialize this in-class?

Reply:
Old habits die hard...